### PR TITLE
Minor updates to core/generate_ssh_keys and batchai/_generate_ssh_keys

### LINKED ISSUE #7267: Remove redundant SSH public file checks in callers of core/keys.py's generate_ssh_keys
In #7105, code to check if an SSH public key file exists were added to generate_ssh_keys in `src/azure-cli-core/azure/cli/core/keys.py`. However, the callers of this function essentially make this check. So we should update them and remove these checks.

See:

-  `_generate_ssh_keys()` in `src/command_modules/azure-cli-batchai/azure/cli/command_modules/batchai/custom.py`

- `validate_ssh_key()` in `src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/_validators.py`

- `validate_ssh_key()` in `src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py`
---

### Environment summary

N/A

```
```

### PR DESCRIPTION
**Pull request by adewaleo**
Minor updates to core/keys/generate_ssh_keys and batchai/custom/_generate_ssh_keys.

Closes #7267.

Did not modify validate_ssh_key in the vm or acs module because these methods make checks that also handle the situation where the  --generate-ssh-key option is not supplied.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/command_modules/azure-cli-batchai/azure/cli/command_modules/batchai/custom.py [hunk 1]
```diff
@@ -605,15 +605,9 @@ def _generate_ssh_keys():
...
-    if os.path.exists(private_key_path) and os.path.exists(public_key_path):
-        logger.warning('Reusing existing ssh public key from ~/.ssh')
-        return
-    if os.path.exists(private_key_path):
-        logger.warning('SSH private key id_rsa exists but public key is missing. Please export the public key.')
-        return
     keys.generate_ssh_keys(private_key_path, public_key_path)
-    logger.warning('SSH key files id_rsa and id_rsa.pub have been generated under ~/.ssh to allow SSH access to the '
-                   'nodes. If using machines without permanent storage, back up your keys to a safe location.')
+    logger.warning('Attempted to find or generate SSH key files id_rsa and id_rsa.pub under ~/.ssh to allow SSH access '
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-batchai/azure/cli/command_modules/batchai/custom.py]: I am not so sure we need this warning any more, as azure-cli-core already emits out a warning based the context, right?

Comment (adewaleo) [reply src/command_modules/azure-cli-batchai/azure/cli/command_modules/batchai/custom.py]: Yes it does in most cases. However, with this change no message will be printed if a new private key file is generated. In the past there was always a message printed. Still, this should be expected as the help text for `--generate-ssh-keys` in `az batchai cluster/file-server create -h`  says: "Generate SSH public and private key files in ~/.ssh directory (if missing)."

Comment (adewaleo) [reply src/command_modules/azure-cli-batchai/azure/cli/command_modules/batchai/custom.py]: I will go ahead and remove this warning.

**Comment by adewaleo** [comment]
@yugangw-msft is this good to merge?

**Pull request merged**